crates/bevy_ecs/src/world/mod.rs

```diff
@@ -486,50 +492,55 @@ impl World {
     /// # let mut world = World::default();
     /// // Register B and C as required by A and C as required by B.
     /// // A requiring C directly will overwrite the indirect requirement through B.
-    /// world.register_required_components::<A, B>();
-    /// world.register_required_components_with::<B, C>(|| C(1));
-    /// world.register_required_components_with::<A, C>(|| C(2));
+    /// world.register_related_components::<A, B>();
+    /// world.register_related_components_with::<B, C>(|| C(1));
+    /// world.register_related_components_with::<A, C>(|| C(2));
     ///
     /// // Duplicate registration! Even if the constructors were different, this would fail.
-    /// assert!(world.try_register_required_components_with::<B, C>(|| C(1)).is_err());
+    /// assert!(world.try_register_related_components_with::<B, C>(|| C(1)).is_err());
     ///
     /// // This will implicitly also insert B with its Default constructor and C
     /// // with the custom constructor defined by A.
     /// let id = world.spawn(A).id();
     /// assert_eq!(&B(0), world.entity(id).get::<B>().unwrap());
     /// assert_eq!(&C(2), world.entity(id).get::<C>().unwrap());
     /// ```
-    pub fn try_register_required_components_with<T: Component, R: Component>(
+    pub fn try_register_related_components_with<T: Component, R: Component>(
         &mut self,
         constructor: fn() -> R,
-    ) -> Result<(), RequiredComponentsError> {
+        relatedness: Relatedness,
+    ) -> Result<(), RelatedComponentsError> {
         let requiree = self.register_component::<T>();
 
         // TODO: Remove this panic and update archetype edges accordingly when required components are added
         if self.archetypes().component_index().contains_key(&requiree) {
-            return Err(RequiredComponentsError::ArchetypeExists(requiree));
+            return Err(RelatedComponentsError::ArchetypeExists(requiree));
         }
 
         let required = self.register_component::<R>();
 
         // SAFETY: We just created the `required` and `requiree` components.
         unsafe {
-            self.components
-                .register_required_components::<R>(required, requiree, constructor)
+            self.components.register_related_components::<R>(
+                required,
+                requiree,
+                constructor,
+                relatedness,
+            )
         }
     }
 
-    /// Retrieves the [required components](RequiredComponents) for the given component type, if it exists.
-    pub fn get_required_components<C: Component>(&self) -> Option<&RequiredComponents> {
+    /// Retrieves the [related components](RelatedComponents) for the given component type, if it exists.
+    pub fn get_required_components<C: Component>(&self) -> Option<&RelatedComponents> {
```

Comment:
Do there need to be new getters here to expose included and suggested components?